virtool/fake/next.py

```diff
@@ -157,3 +171,33 @@ async def create(
             return await self.layer.users.get(user.id)
 
         return user
+
+
+class UploadsFakerPiece(DataFakerPiece):
+    model = Upload
+
+    async def create(
+        self,
+        user: User,
+        with_file: bool = False,
+        upload_type: str = "reads",
+        name: str = "test.fq.gz",
+        reserved: bool = False,
+    ):
+
+        if upload_type not in UploadType.to_list():
+            upload_type = "reads"
+
+        upload = await self.layer.uploads.create(name, upload_type, reserved, user.id)
+
+        if with_file:
+            config = getattr(self.layer.uploads, "_config")
+            file_path = config.data_path / "files" / upload.name_on_disk
+            size = await naive_writer(fake_file_chunks(file_path), file_path)
+
+        if not with_file:
+            size = 1
```

Comment:
Also, use random `int` from Faker for `size`. One byte is not realistic.
```suggestion
        size = 1
        
        if with_file:
            config = getattr(self.layer.uploads, "_config")
            file_path = config.data_path / "files" / upload.name_on_disk
            size = await naive_writer(fake_file_chunks(file_path), file_path)
```